File: spec/requests/managing_dogs_spec.rb
require 'spec_helper'

describe 'Managing Dogs' do

  it 'should be able to add a dog' do
    Dog.count.should == 0
    visit root_path
    click_link 'Dogs'
    click_link 'Add a new dog'

    fill_in_fields :dog,
      :name  => 'Rover',
      :breed => 'Golden Retriever'
    click_button 'Add Dog'

    # Dog.count.should == 1
    Dog.first.name.should  == 'Rover'
    Dog.first.breed.should == 'Golden Retriever'
  end

  it 'should be able to add a dog with javascript', :js => true do
    Dog.count.should == 0
    visit root_path
    click_link 'Dogs'
    click_link 'Add a new dog'

    fill_in_fields :dog,
      :name  => 'Rover',
      :breed => 'Golden Retriever'
    click_button 'Add Dog'

    # Dog.count.should == 1
    Dog.first.name.should  == 'Rover'
    Dog.first.breed.should == 'Golden Retriever'
  end

  it 'can correct validation errors'
  # ... 

end

File: TestDrivenDogs.Specs/Web/ManagingDogsSpec.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;

namespace TestDrivenDogs.Specs.Web {

	[TestFixture]
	public class ManagingDogsSpec : WebSpec {

		[Test]
		public void can_view_list_of_dogs() {
			Visit("/");
			Click("Dogs");
			Page.Body.ShouldNotContain("Awesome Rover");

			f.Dog.Create(new { Name = "Awesome Rover" });

			Refresh();
			Page.Body.ShouldContain("Awesome Rover");
		}

		[Test]
		public void can_create_dog() {
			var db = new DogsContext();
			db.Dogs.Count().ShouldEqual(0);

			Visit("/");
			Click("Dogs");
			Click("Add Dog");
			FillIn("Name", "Awesome Spot");
			FillIn("Breed", "Golden Retriever");
			FillIn("VetId", "5");
			FillIn("RegisteredAt", DateTime.Now.ToString());
			FillIn("UniqueId", Guid.NewGuid().ToString());
			Click("Create");

			db.Dogs.Count().ShouldEqual(1);
			db.Dogs.First().Name.ShouldEqual("Awesome Spot");
		}

		[Test]
		public void cant_create_invalid_dog() {
			var db = new DogsContext();
			db.Dogs.Count().ShouldEqual(0);

			Visit("/");
			Click("Dogs");
			Click("Add Dog");
			FillIn("Name", string.Empty);
			Click("Create");

			db.Dogs.Count().ShouldEqual(0);
			Page.Body.ShouldContain("The Name field is required.");
		}
	}
}
